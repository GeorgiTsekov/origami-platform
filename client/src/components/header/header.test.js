import React from 'react';
import renderer from 'react-test-renderer';
import Header from './header'
import TestingEnvironment from '../../test-utils/router';
import getNavigation from '../../utils/navigation';

describe('Header Component', () => {
    it('return authenticated routes from getNavigation', () => {
        const links = getNavigation({
            loggedIn: true,
            id: '123'
        });
        const authLinks = [
            {
                title: "Publications",
                link: "/"
            },
            {
                title: "Share your throughts",
                link: "/share"
            },
            {
                title: "Profile",
                link: `/profile/123`
            }
        ];
        console.log(authLinks);
        expect(links).toStrictEqual(authLinks);
    });
    it('shoult render authenticated routes', () => {
        const tree = renderer.create(
            <TestingEnvironment value={{
                user: {
                    loggedIn: true,
                    id: '123'
                }
            }}>
                <Header />
            </TestingEnvironment>

        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('shout render non-authenticated routes', () => {
        const tree = renderer.create(
            <TestingEnvironment value={{
                user: {
                    loggedIn: false
                }
            }}>
                <Header />
            </TestingEnvironment>

        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});