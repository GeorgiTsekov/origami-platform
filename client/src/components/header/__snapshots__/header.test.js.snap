// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component shoult render authenticated routes 1`] = `
<header
  className="navigation"
>
  <img
    alt=""
    className="logo"
    src="white-origami-bird.png"
  />
  <li
    className="header-list-item"
    data-test-id="link-Publications"
  >
    <a
      className="header-link"
      href="/"
      onClick={[Function]}
    >
      Publications
    </a>
  </li>
  <li
    className="header-list-item"
    data-test-id="link-Share your throughts"
  >
    <a
      className="header-link"
      href="/share"
      onClick={[Function]}
    >
      Share your throughts
    </a>
  </li>
  <li
    className="header-list-item"
    data-test-id="link-Profile"
  >
    <a
      className="header-link"
      href="/profile/123"
      onClick={[Function]}
    >
      Profile
    </a>
  </li>
</header>
`;

exports[`Header Component shout render non-authenticated routes 1`] = `
<header
  className="navigation"
>
  <img
    alt=""
    className="logo"
    src="white-origami-bird.png"
  />
  <li
    className="header-list-item"
    data-test-id="link-Publications"
  >
    <a
      className="header-link"
      href="/"
      onClick={[Function]}
    >
      Publications
    </a>
  </li>
  <li
    className="header-list-item"
    data-test-id="link-Register"
  >
    <a
      className="header-link"
      href="/register"
      onClick={[Function]}
    >
      Register
    </a>
  </li>
  <li
    className="header-list-item"
    data-test-id="link-Login"
  >
    <a
      className="header-link"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </li>
</header>
`;
